### DISCUSSION TITLE
Always build vcpkg-tool from souce

### DISCUSSION THREAD

**Discussion by msclock** [microsoft/vcpkg#42248] [Q&A] [answered]
I encouter GLIBC-not-found problems such as `/usr/lib/libc.so.6: version `GLIBC_2.14’ not found` on ~~x86 linux~~ x86_x64 linux. In bootstrap.sh, it always chooses to download vcpkg-tool instead of build from source.

And are there any ways to implement the behaviour?

**Comment by bansan85** [comment, marked as answer]
Take a look at https://github.com/microsoft/vcpkg/blob/834977918592e4f5d5ec5fe48ba338c608faf9f9/scripts/bootstrap.sh#L172-L188

You can't decide if you want to build from source or not. But if you are on x86 linux (and not x86_64), you should always build from source.

**Reply by msclock** [reply]
@bansan85 Sorry for my wrong tying.😓 I reviewed the failed logs https://github.com/msclock/cppcheck-wheel/actions/runs/11927833904/job/33243849191#step:3:399. And it is a x86_64 linux.

it runs on the old image `quay.io/pypa/manylinux2010_x86_64:2022-08-05-4535177` of cibuildwheel in ci.

> You can't decide if you want to build from source or not. 

Yep, the scripts always set it ON for vcpkgDownloadTool. It seems it is a glic download problem that it is not compatible to `GLIBC_2.14` from vcpkg baseline.

**Reply by msclock** [reply]
Finally, I switch to a newer image x86_x64 image. It works.

Anyway, thanks for your explanation.

**Reply by msclock** [reply]
Another workaround is to patch the bootstrap script and bootstrap vcpkg manually.  See
https://github.com/serious-scaffold/ss-cpp/blob/3a83b786e18d2f6df29bad849ab266dc372f31ab/cmake/vcpkg/bootstrap/vcpkg_bootstrap.cmake#L66-L96

**Reply by bansan85** [reply]
I just needed to use vcpkg with Ubuntu 12.04.

I needed to use the following patch: [vcpkg.patch](https://github.com/user-attachments/files/18140739/vcpkg.patch)

**Reply by msclock** [reply]
Currently, it does need patches
